### ISSUE TITLE
Log information in a progressive manner rather than gate details behind `--no-silent`

### ISSUE THREAD

**Issue by RA80533** [angular-schule/angular-cli-ghpages#129]
I encountered an issue with deploying. I didn't know what caused it until I enabled `--no-silent`:

```console
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
```

I do not believe the necessity of the flag is warranted for such scenarios.

**Comment by JohannesHoppe** [comment]
@RA80533 I completely agree with you. I used to hide the output because some years ago it was still very common to use the username and password directly to authenticate git. Nowadays, hopefully no one is doing this anymore.

In addition, CI environments like Github Actions and Circle CI hide the secrets in the output. Therefore, we should also be safe here.

It will certainly be a great advantage to change the defaults because then we will have to provide less support by telling everybody to use `--no-silent`. I will add this breaking change to `-rc.3`.
